test/common/buffer/buffer_speed_test.cc

```diff
@@ -207,14 +218,10 @@ BENCHMARK(bufferMovePartial)->Arg(1)->Arg(4096)->Arg(16384)->Arg(65536);
 static void bufferReserveCommit(benchmark::State& state) {
   Buffer::OwnedImpl buffer;
   for (auto _ : state) {
-    constexpr uint64_t NumSlices = 2;
-    Buffer::RawSlice slices[NumSlices];
-    uint64_t slices_used = buffer.reserve(state.range(0), slices, NumSlices);
-    uint64_t bytes_to_commit = 0;
-    for (uint64_t i = 0; i < slices_used; i++) {
-      bytes_to_commit += static_cast<uint64_t>(slices[i].len_);
-    }
-    buffer.commit(slices, slices_used);
+    UNREFERENCED_PARAMETER(_);
+    auto size = state.range(0);
+    Buffer::Reservation reservation = buffer.reserveApproximately(size);
+    reservation.commit(size);
```

Comment:
the size of the reservation is ignored, it is possible for commit to be called with a size that is larger than the reservation.  this will happen if for example size > 128kb or an unusual size is passed in like 16500 bytes.